tracer/src/Datadog.Trace/Iast/ReturnedHeadersAnalyzer.cs

```diff
@@ -39,6 +41,127 @@ internal static void Analyze(IHeaderDictionary responseHeaders, IntegrationId in
         AnalyzeXContentTypeOptionsVulnerability(responseHeaders, integrationId, serviceName, responseCode);
         AnalyzeStrictTransportSecurity(responseHeaders, integrationId, serviceName, responseCode, protocol);
         AnalyzeUnvalidatedRedirect(responseHeaders, integrationId);
+        AnalyzeHeaderInjectionVulnerability(responseHeaders, integrationId);
+    }
+
+#if NETFRAMEWORK
```

Comment:
We should begin segregating this in different files (partial class files should do), because the file is beginning to grow too much, IMHO

Reply:
I agree, I suggest extracting this as two different methods, and then extracting the body of the method to a shared helper function

Reply:
Yes, we should consider refactoring the whole file since it has grown quite a lot recently. I have created a Jira task for this small refactor.